src/persist/src/storage.rs

```diff
@@ -16,8 +16,46 @@ use std::str::FromStr;
 
 use abomonation_derive::Abomonation;
 use async_trait::async_trait;
+use futures_executor::block_on;
 
 use crate::error::Error;
+use crate::indexed::encoding::BlobMeta;
+
+/// Sanity check whether we can decode the Blob's persisted meta object.
+///
+/// TODO: this is a hack and we will need to get rid of this once we have a
+/// proper backwards compatibility policy.
+pub fn check_meta_version<B: Blob>(b: &mut B) -> Result<(), Error> {
```

Comment:
the name of this should be much more loud that it deletes all the data if the version doesn't match

Reply:
came up with `check_meta_version_maybe_delete_data` lemme know if you have a better name in mind as this feels clunky

Reply:
clunky is fine :)